### PR TITLE
cleanup bignumber handling, formatting

### PR DESCRIPTION
**Pull request by aquaticone**

### CODE REVIEW

File: src/components/Asset/AssetDetails.tsx [hunk 1]
```diff
@@ -18,87 +16,73 @@ export type AssetDetailsProps = {
...
-      {!isReady || !isConnected || !address ? (
-        <>—</>
-      ) : (
-        <Currency
-          amount={balance?.value ?? BigNumber.from(0)}
-          decimals={balance?.decimals ?? 18}
-          abbreviate={abbreviate}
-        />
-      )}
+    <Skeleton isLoading={isLoading}>
```
Comment (vanillaHill) [inline src/components/Asset/AssetDetails.tsx:54]: Do we no longer display "-" for non-connected users?

File: src/components/VaultRow/lib/VaultDepositForm.tsx [hunk 2]
```diff
@@ -170,7 +166,7 @@ const VaultDepositForm: FC<VaultProps> = (props) => {
       </h2>
       <FormProvider {...form}>
         <TokenForm
-          isError={prepareDeposit.isError || prepareDepositUnderlying.isError}
+          isError={prepareDeposit.isError}
```
Comment (vanillaHill) [inline src/components/VaultRow/lib/VaultDepositForm.tsx:169]: why prepareDepositUnderlying.isError was deleted?

**Review by vanillaHill — CHANGES_REQUESTED** [review]

**Review by vanillaHill — APPROVED** [review]

File: src/components/VaultRow/lib/VaultDepositForm.tsx (hunk 2 above)
Comment (aquaticone) [reply src/components/VaultRow/lib/VaultDepositForm.tsx:169]: This should not have been deleted.

File: src/components/Asset/AssetDetails.tsx (hunk 1 above)
Comment (aquaticone) [reply src/components/Asset/AssetDetails.tsx:54]: We do its right there on line 61

**Pull request merged**